code/Chapter5/employee_table.h

```diff
@@ -0,0 +1,32 @@
+#ifndef employee_table_h
+#define employee_table_h
+
+#include <vector>
+
+template <class GenericEmployee>
+class EmployeeTable
+{
+private:
+    std::vector<GenericEmployee> employees;
+
+public:
+    // Member function to add employees into the table.
+    void addEmployee(const GenericEmployee &employee);
+
+    // Member function to remove employees into the table.
+    void removeEmployee(const std::string &employeeName);
+
+    // Member function to display information (names) of all employees in the table.
+    void displayEmployeesName() const;
+
+    // Function to check if the table is empty.
+    bool isEmpty() const;
+
+    // Function to get the number of entries in the table.
+    int getEntryCount() const;
+
+    // Member function to change the bonus to all employees.
+    void bumpSalaryBonus(const double newBonus);
+};
+
+#endif /* employee_table_h */
```

Comment:
```suggestion
#ifndef employee_table_h
#define employee_table_h

#include <vector>

template <class GenericEmployee>
class EmployeeTable
{
private:
    // See the note number 2 given in exercise_with_mocks on why we are using pointers here.
    std::vector<GenericEmployee*> employees;

public:
    // Member function to add employees into the table.
    void addEmployee(GenericEmployee* employee)
    {
        employees.push_back(employee);
    }

    // Other functions may or may not require changes.
    // For brevity, we are using only two functions.

    // Member function to change the bonus to all employees.
    void bumpSalaryBonus(const double newBonus)
    {
        for (auto& employee : employees)
        {
            employee->setIncreasedBasicBonus(newBonus);
        }
    }
};

#endif /* employee_table_h */
```

@dalonsoa Sorry for disturbing you. If you want, I can do this on my own but I did not want to hijack your PR as you have done all the work for this chapter. With my new suggestion, you would only need to accept a few things which should not be a major burden. 

I have tried to keep the same goal which you were trying to achieve. I just deleted a few lines which are not needed for the exercise and makes the task a bit longer.

There were a few gotchas which we figured out. We have now mentioned them in the comments. Also, many thanks to @alexdewar for helping.